ios/RNCAsyncStorage.m

```diff
@@ -156,6 +162,52 @@ static dispatch_queue_t RCTGetMethodQueue()
   return error ? RCTMakeError(@"Failed to delete storage directory.", error, nil) : nil;
 }
 
+
+NSString *OldStorageDirectory = @"RNCAsyncLocalStorage_V1";
+/**
+ * Creates an NSException used during Storage Directory Migration.
+ */
+static NSException *RCTStorageDirectionMigrationException(NSString *reason, NSError *error) {
+  NSMutableString *reasonString = [[NSMutableString alloc] initWithString:reason];
+  [reasonString appendString:@" - "];
+  [reasonString appendString:[error description]];
+  return [[NSException alloc] initWithName:@"RCTStorageDirectoryMigrationFailure" reason:reasonString userInfo:nil];
+}
+
+/**
+ * This check is added to make sure that anyone coming from pre-1.2.2 does not lose cached data.
+ * Data is migrated from the "RNCAsyncLocalStorage_V1" directory to the "RCTAsyncLocalStorage_V1" directory.
+ */
+static void RCTStorageDirectoryMigrationCheck() {
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    NSError *error;
+    BOOL isDir;
+    // If the old directory exists, it means we need to migrate data to the new directory
+    if ([[NSFileManager defaultManager] fileExistsAtPath:RCTCreateStorageDirectoryPath(OldStorageDirectory) isDirectory:&isDir] && isDir) {
+      // Check if the new storage directory location already exists
+      BOOL newStorageDirectoryExists = [[NSFileManager defaultManager] fileExistsAtPath:RCTGetStorageDirectory() isDirectory:&isDir];
+      if (newStorageDirectoryExists) {
+        // If the new storage directory location already exists, remove existing directory
+        newStorageDirectoryExists = !([[NSFileManager defaultManager] removeItemAtPath:RCTGetStorageDirectory() error:&error]);
```

Comment:
I can't help but think that we might lose data here depending on which version you're migrating from. Would it be better to check for last modified folder and migrate that one? I think the `manifest.json` file is updated with every write and can be used as a pretty good indicator.

Reply:
This does sound a lot safer, I'll have a look at the manifest file.